modules/swagger-codegen/src/main/resources/swift3/AlamofireImplementations.mustache

```diff
@@ -127,11 +136,15 @@ open class AlamofireRequestBuilder<T>: RequestBuilder<T> {
                 }
             })
         } else {
-            let request = makeRequest(manager: manager, method: xMethod!, encoding: encoding, headers: headers)
-            if let onProgressReady = self.onProgressReady {
-                onProgressReady(request.progress)
+            do {
+                let request = try makeRequest(manager: manager, method: xMethod!, encoding: encoding, headers: headers)
+                if let onProgressReady = self.onProgressReady {
+                    onProgressReady(request.progress)
+                }
+                processRequest(request: request, managerId, completion)
+            } catch {
+                completion(nil, ErrorResponse.HttpError(statusCode: 500, data: nil, error: error))
```

Comment:
Is 500 the right status code here?

Reply:
y that looks right

Reply:
I'm not sure about returning a response code if this didn't come from the server. It could be confusing / misleading.

Reply:
I agree and I'm not really sure HttpError actually make since here. Changing how the error system would be a bigger change?

Also I choose 500 since that was the closest to what processRequest would have done. I think this is the response you would have gotten in the old code. 
`ErrorResponse.HttpError(statusCode: response.response?.statusCode ?? 500, data: response.data, error: response.result.error!)`

Reply:
@jgavris hmm we do send the 500 at the end of processRequest in AlamofireImplementations as well - maybe we should avoid doing that - is there a better type of ErrorResponse to return?

Reply:
I see that...yes we should improve it. Maybe we can add a `case ClientError` to `ErrorResposne` enum?

Reply:
For other API client (e.g. PHP), we set the status code to 0 in the API exception object/class if the issue happens in the client side (e.g. network issue). Maybe we can follow the same convention in the Swift client.